tuned/plugins/plugin_cpu.py

```diff
@@ -203,11 +232,31 @@ def _instance_apply_static(self, instance):
 	def _instance_unapply_static(self, instance, full_rollback = False):
 		super(CPULatencyPlugin, self)._instance_unapply_static(instance, full_rollback)
 
+		if len(self._reenable_thread_siblings) > 0:
+			log.info("re-enabling thread siblings: %s" % str(self._reenable_thread_siblings))
+			for c in self._reenable_thread_siblings:
+				self._cmd.write_to_file("/sys/devices/system/cpu/cpu%d/online" % c, "1")
+			self._reenable_thread_siblings= []
+			self._storage.set(self._thread_siblings_storage_key, self._reenable_thread_siblings)
+
 		if instance._first_instance and self._has_intel_pstate:
 			self._set_intel_pstate_attr("min_perf_pct", self._min_perf_pct_save)
 			self._set_intel_pstate_attr("max_perf_pct", self._max_perf_pct_save)
 			self._set_intel_pstate_attr("no_turbo", self._no_turbo_save)
 
+	def _instance_verify_static(self, instance, ignore_missing, devices):
+		ret1 = super(CPULatencyPlugin, self)._instance_verify_static(instance, ignore_missing, devices)
+		ret2 = True
+		offline_cpus = []
+		for c in self._reenable_thread_siblings:
```

Comment:
Could you skip the thread_siblings verification if the `len(self._disable_thread_siblings_val) == 0`? I.e. do not verify it if user didn't request it. It may be more readable and maybe is saves some CPU cycles if the feature is not used.

Reply:
This is not blocker.

Reply:
I've restructured some of the logic, and also tried to make the calcucation of cpus that will be disabled a bit more explicit.